### PR TITLE
Quick report system for security officers

### PR DESCRIPTION
**Pull request by twilightwanderer**
## About The Pull Request

Adds a special skill for station security officers, as well as a system of quick radio reports for security headphones. The security officer will be able to call up a special earpiece menu with the choice of a short report and say it to the security channel with his location. This is formalized through a special chip as a skill similar to engineers' knowledge of the wires. 

The following reports are currently available:
1. Combat Actions
- Pursuing the perpetrator.
- I'm fighting.
- I'm hurt.
- Taking combat losses.
- Retreat

2. Detecting
- Station damage.
- Fire.
- Depressurization.
- Criminal trail.
- Corpse.
- Perpetrator.
- Dangerous perpetrator.

3. Status
- Positive
- Negative
- My location
- Busy
- Available

4. Support Request
- Requesting a detective
- Requesting combat support
- Requesting Engineering assistance
- Requesting Medical assistance

## Why It's Good For The Game

Sometimes the game as an officer is so dynamic that there is simply not enough time to write in a chat call for help before you get killed. With this system, I want to give officers at least some chance to cooperate in dynamic battles by giving them the ability to send basic messages to the security channel at the push of a button. I have framed this as a combat skill learned in special training before arriving at the station, sort of like knowing special radio codes for real policemen. I also tried to strike a balance by making the reports minimal, without describing the nature of the criminals and crimes. 

## Changelog

:cl:
add: quick report system for security officers
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
looks neat but im not sure how to feel about players speaking game generated lines. do you think we could "objectify" it a bit more by making the radio the thing reporting on whatever?
also seems like it should have a cooldown

**Comment by twilightwanderer** [comment]
> cooldown

I understand that this is a rather controversial innovation and I would really like to listen to feedback from other players. On the server on which I play, security officers often lose because of a basic lack of cooperation and the ability to respond quickly to the threat. For example, there is often a situation where an officer notices a threat and while he will write about it in the channel, the threat will run to the other end of the station. I wanted to make something like the real radio codes that police officers from different countries use in their work. As for the generated text, I could frame it as random numbers. It would look like the officer was saying a random number like 10-21301, 11-12442, and the earpiece would decode it with a robotic voice into words. But that doesn't seem fair to me to the antagonists. As for cooldown, I wanted to implement this feature, but I don't know what delay time would be optimal yet.

**Comment by Mothblocks** [comment]
I can't tell if I find this funny or not but barring that, this should be on sec headsets themselves, and not a skillchip. That way, the headsets themselves can be what makes a digitized voice.

Human speech doesn't make a lot of sense considering a lot of the time you're using something like this, you might be in combat, and so your character would not be calmly saying Requesting assistance in Medbay Primary Treatment Center as they're getting their head sliced open.

**Comment by wesoda25** [comment]
Really cool but I sort of liked how you had to choose between defending yourself/running or calling for help. I guess it won't change things that much though since this'll still take time, just a bit less.

**Comment by Cheshify** [comment]
It'd be cool if it was part of the security officer's equipment so deputized officers don't need to go to the library to be able to do their job at full efficiency. People suggested headsets, I think that'd be cool.

**Comment by twilightwanderer** [comment]
> It'd be cool if it was part of the security officer's equipment so deputized officers don't need to go to the library to be able to do their job at full efficiency. People suggested headsets, I think that'd be cool.



> It'd be cool if it was part of the security officer's equipment so deputized officers don't need to go to the library to be able to do their job at full efficiency. People suggested headsets, I think that'd be cool.

I have provided for this and all security personnel, with the exception of the detective, have a default skill similar to that of an engineer. Only I did not extend this to staff members who can be arranged by the head of staff in the round. I don't know if it's worth adding a skill chip to the sale for them yet.

**Comment by cacogen** [comment]
>I think this approach is wrong, because it allows anyone using the mask to radiate crime codes.

It's good if something in-game has counterplay and isn't objective/infallible.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Please make it a part of sec headsets, and use a digitized voice.

**Comment by JohnFulpWillard** [comment]
I don't see the point of the Status tab, but I would like the ability to request a Lawyer as you can for Detective.

**Comment by twilightwanderer** [comment]
> Really cool but I sort of liked how you had to choose between defending yourself/running or calling for help. I guess it won't change things that much though since this'll still take time, just a bit less.
> 
> As an aside, imo you should make this as barebones as possible and only include very basic phrases. Like, "need backup", "situation resolved", "pursuing the perp", maybe 2-3 more. Personally I think only combat options should be available, this is a roleplaying game after all and if there no sense of urgency players should absolutely be using their own words.
> 
> Also also, can you mention how to use it in changelog?

In creating the phrases I was guided by the fact that such a system will be used specifically in battle or in some urgent case. Some of the phrases serve directly to inform about the conduct of combat, some inform about the danger, and some give a request for any help. As I like to believe, this system gives a tactical element to the players. For example, a Varden sitting in a brig listening to reports like: Depressurization, I'm fighting, I'm hurt, should realize that the officer is fighting in an airless environment and is wounded. He should send in reinforcements, medics and engineers in spacesuits.

I'm not quite sure how to document changelog. I couldn't find the right tag. Is it documented in a separate document?

**Comment by twilightwanderer** [comment]
> By the way, if you're requesting engineering or medical assistance, wouldn't it make more sense to send that over common or the specific frequency if available (or even to the department's request console?)

I do this with the hope that some available officer hearing this request from a colleague in combat will pass it on the general channel.

**Comment by cacogen** [comment]
I think in most cases it's probably unlikely

**Comment by realthomas69** [comment]
this is a roleplaying game and we dont need game generated voice lines. just use your keyboard dude, insane

**Comment by twilightwanderer** [comment]
> I don't see the point of the Status tab, but I would like the ability to request a Lawyer as you can for Detective.

Hmm, in about two years of playing, I've never seen the need for an urgent request for a lawyer. Okay, I'll add it.

**Comment by realthomas69** [comment]
If it was changed to just say [crime codes,](https://tgstation13.org/wiki/Space_Law#Crime_Codes_Quick_Reference) and you could like mouse over them to see what they mean, i think that would be pretty rad

**Comment by cacogen** [comment]
Isn't there already a bubble in the bottom right you can click on to bring up a list of preset things to say?

**Comment by twilightwanderer** [comment]
> this is a roleplaying game and we dont need game generated voice lines. players should have talk to eachother and use their own words, lol

The fastest keyboard cowboy in Wild Space. His hands were ahead of the Syndicate operatives' rifle shots. He could fight the Revenant, the Revolution, the Mafia, and the Tulbox assistant simultaneously, while confessing his feelings for the girl he loved, ordering pizza to the department, and insulting the captain.

**Comment by twilightwanderer** [comment]
I'm not trying to take away your role-playing. I'm just trying to simplify life in combat situations where stopping to write something is to die.

**Comment by wesoda25** [comment]
That's a decision people have to make though. Fight for your life and potentially die quiet, or give up your edge in combat and alert the other members of security.

**Comment by StoneMason** [comment]
This PR sort of feel like it’s infantalizing security. If you’re a security officer, you should be communicating with your team, to automate that would just drain the already low skill cap of security.

Though, this could be a cool upgrade for sec masks (hailers? idk) locked via research or BEPIS.

**Comment by twilightwanderer** [comment]
> If it was changed to just say [crime codes,](https://tgstation13.org/wiki/Space_Law#Crime_Codes_Quick_Reference) and you could like mouse over them to see what they mean, i think that could be interesting



> If it was changed to just say [crime codes,](https://tgstation13.org/wiki/Space_Law#Crime_Codes_Quick_Reference) and you could like mouse over them to see what they mean, i think that could be interesting

This would be logical, but these codes depend on the game policies of specific servers. On official tgstation servers these codes are used, on servers that use TG builds other codes are used. There are also a bunch of upstreams like Skyrat, which also use TG as downstream. This unfortunately must also be taken into account.

**Comment by Horatio22** [comment]
I really like this and think it's a good idea to have a way for sec officers to quickly communicate with each other in the middle of combat. I am a little curious as to why there are so many things you can request though, and I think some of the concerns about people just using the request menu instead of typing out a response are valid. I think this would function better and discourage using this in place of typing out a response if it were limited to asking for emergency support (e.g. without the detective, engineer, and anything else not immediately necessary in combat).

**Comment by cacogen** [comment]
I like that you've included reports security can make that in most cases probably aren't time-constrained, because their existence will remind people to make security reports they may not think to make otherwise.

File: code/game/objects/items/devices/radio/headset.dm [hunk 1]
```diff
@@ -355,3 +366,55 @@ GLOBAL_LIST_INIT(channel_tokens, list(
...
+				"Requesting a detective" = image(icon = 'icons/mob/actions/action_security.dmi', icon_state = "detective_support"),
+				"Requesting combat support" = image(icon = 'icons/mob/actions/action_security.dmi', icon_state = "combat_support"),
+				"Requesting Engineering assistance" = image(icon = 'icons/mob/actions/action_security.dmi', icon_state = "engineering_assistance"),
+				"Requesting Medical assistance" = image(icon = 'icons/mob/actions/action_security.dmi', icon_state = "medical_assistance")
+			)
+	var/message = show_radial_menu(user, user, options)
+	if(!message)
+		return FALSE
+	message = "[message]. [get_area(user)]."
+	radio.talk_into(user, "[message]", "Security")
```
Comment (ATH1909) [inline code/game/objects/items/devices/radio/headset.dm:419]: what happens if your headset lacks a security encryption key?

**Comment by ATH1909** [comment]
inB4 antags start stealing blueprints so that they can rename the areas they plan to gank sec officers in to make their reports refer to a wrong/funny location name (i.e. renaming Dorms to "Medbay", an IC-banned word, or "Boy, I sure do hate minorities.")

**Comment by LemonInTheDark** [comment]
> That's a decision people have to make though. Fight for your life and potentially die quiet, or give up your edge in combat and alert the other members of security.

I agree with this post
In your explanation, you said this:

> Sometimes the game as an officer is so dynamic that there is simply not enough time to write in a chat call for help before you get killed. With this system, I want to give officers at least some chance to cooperate in dynamic battles by giving them the ability to send basic messages to the security channel at the push of a button. I have framed this as a combat skill learned in special training before arriving at the station, sort of like knowing special radio codes for real policemen. I also tried to strike a balance by making the reports minimal, without describing the nature of the criminals and crimes.

You have not done anything to explain why the current state of things is bad, or why making communication easier is a good thing, outside of just discussing calling for help, which frankly seems like a good thing not a bad thing to me.

My gut reaction to making this easier just for its own sake is speed close, but that seems a bit rude. Please convince me why freer communication is something we even want.

**Comment by theta060** [comment]
because sec needs to be stronger for some reason if sec uses their brain and tools right they can already overpower most antags solo the only exception is cult hardstun bullshit getting JFKed by a nukie 50 cal and blob other then that its truly a skill issue

File: code/modules/library/skill_learning/job_skillchips/security_officer.dm [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/obj/item/skillchip/job/security
+	name = "SEC-Ten skillchip"
+	desc = "Newspeak Dictionary for the staff of the Ministry of Peace Nanotrasen."
+	auto_traits = list(TRAIT_KNOW_SECURITY_REPORTS)
+	skill_name = "Police Newspeak"
+	skill_description = "Knowledge of security radio reports."
+	skill_icon = "dumbbell"
+	activate_message = "<span class='notice'>The musical instruments in your head start playing the main track of the Police Academy, and your nose picks up the smell of every donut at the station.</span>"
+	deactivate_message = "<span class='notice'>The Police Adademia track is replaced by a cacophony, and the pleasant smell of donuts interrupts the runny nose.</span>"
```
Comment (UnderAGeode) [inline code/modules/library/skill_learning/job_skillchips/security_officer.dm:9]:
```suggestion
	activate_message = "<span class='notice'>Musical instruments in your head start playing the main track of the Police Academy, and your nose picks up the smell of every donut on the station.</span>"
	deactivate_message = "<span class='notice'>The Police Academy track is replaced by a cacophony, and the pleasant smell of donuts is interrupted by a runny nose.</span>"
```

**Comment by GuillaumePrata** [comment]
> this is a roleplaying game and we dont need game generated voice lines. players should have talk to eachother and use their own words, lol

I agree with this. The idea is interesting tho, it's just giving too much information.
A button on the top that is given by the sec headset and just pings the security channel would be interesting, as you are given no extra information about anything, just that an officer is possibly on danger, it pushes officers to ask the AI for their location, to use clues from the round to guess what is the issue at hand, ask paramedics to help them track their corpse, etc.

Basically, it achieves the goal of stopping officers that die silently and no one ever realizes for 20 minutes without removing the uncertainty of what you will have to deal with if you try to rush after the ping.

Also bonus points for the officers that will press this by mistake and get laughed at by other officers (hopefully in a not rude way...)

**Comment by ATH1909** [comment]
> You have not done anything to explain why the current state of things is bad, or why making communication easier is a good thing, outside of just discussing calling for help, which frankly seems like a good thing not a bad thing to me.
> 
> My gut reaction to making this easier just for its own sake is speed close, but that seems a bit rude. Please convince me why freer communication is something we even want.

something something grod's law

**Comment by Victor239** [comment]
I think it's more interesting as a HoS exclusive item with their unique gas mask.

It makes them harder to assassinate without being a straight combat buff like stronger weapons or armour would, which otherwise just makes them into loot pinatas. 

It also enables them to more quickly and easily direct their Officers to different situations without having to constantly say long sentences whilst in loud volume mode, which can be annoying for both the listener and speaker.

**Comment by Coastfront** [comment]
I think it gives too much info + being way faster.
If im mid fight with a sec officer as a traitor, and they, within 2 button presses, have given precise details where they are, and that a fight is happening, without the added 1 or 2 seconds of typing it out, im fucked almost no matter how fast i go, and they still have a shot at winning.
If it just pinged sec coms that'd be way less strong than "I am in (location) fighting" when sec's only real weakness is being split up and having poor communication, if you make communication even easier, they just get stronger.

**Comment by optimumtact** [comment]
is typing in the game too hard for you?

**Comment by Kapu1178** [comment]
Thanks oranges 
image

**Comment by twilightwanderer** [comment]
> is typing in the game too hard for you?

With the amount of text I've written here trying to explain the meaning of this PR, then yes, apparently the text is very hard for me.

**Comment by optimumtact** [comment]
very factual
